src/app/users/get-user.ts

```diff
@@ -1,65 +1,53 @@
-import {
-  SemanticAttributes,
-  asyncRunInSpan,
-  addAttributesToCurrentSpan,
-  ENDUSER_ALIAS,
-} from "@services/tracing"
+import { SemanticAttributes, asyncRunInSpan } from "@services/tracing"
 import { UsersRepository } from "@services/mongoose"
-import { IpFetcher } from "@services/ipfetcher"
 import { getIpConfig } from "@config/app"
-import { RepositoryError } from "@domain/errors"
+import { IpFetcher } from "@services/ipfetcher"
 
 const users = UsersRepository()
 
-export const getUserForLogin = async ({
+export const updateIpInfo = async ({
```

Comment:
Extracting an `updateIpInfo` from the `getUserForLogin` and calling it asynchronously makes sense IMO - but it should still probably be called as part of the `getUserForLogin` use case. Having the outer adapter layer interact with the `UserRepository` and this use case in different ways is much too granular and introduces back a lot of the coupling and hard to test structure that we are working to get away from.